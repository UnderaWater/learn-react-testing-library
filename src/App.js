import { useEffect, useState } from 'react';
import './App.css';

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');

  const handleClick = () => setToggle(prev => !prev);
  
  useEffect(() => {
    setTimeout(() => {
      setData({})
    }, 300)
  }, [])

  return (
    <div className="App">
      <h1>Hello world</h1>
      <h2 data-testid='value-elem'>{value}</h2>
      {data && <div>text div</div>}
      {toggle === true && <div data-testid='toggle-elem'>toggle</div>}
      <button data-testid='toggle-btn' onClick={handleClick}>Click</button>
      <input onChange={(e) => setValue(e.target.value)} type='text' placeholder='text...' />
    </div>
  );
}

export default App;
