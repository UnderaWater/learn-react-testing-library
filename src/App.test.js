import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

describe('test', () => {
  test('renders learn react link', () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/text.../i)
    expect(helloWorldElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('renders learn react link', () => {
    render(<App />);
    const isElem = screen.queryByText(/hello2/i);
    expect(isElem).toBeNull();
  });
  
  test('async test', async () => {
    render(<App />);
    const data = await screen.findByText(/text div/i);
    expect(data).toBeInTheDocument();
  })

  test('event click', () => {
    render(<App />);
    const btn = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-elem')).toBeNull(); 
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-elem')).toBeNull(); 
  })

  test('event input', () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/text.../i);
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    // fireEvent.input(input, {
    //   target: { value: 'qwerty' }
    // });
    userEvent.type(input, 'qwerty');
    expect(screen.queryByTestId('value-elem')).toContainHTML('qwerty');
  })
});
