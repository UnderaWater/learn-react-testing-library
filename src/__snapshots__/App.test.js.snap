// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders learn react link 1`] = `
<input
  placeholder="text..."
  type="text"
/>
`;
